### PR TITLE
Preparation to splitting GUI and engine

### PR DESCRIPTION
**Pull request by eott-siz**
* Add tests for exporting parameters to OSW files based on default values in the GUI
* Merge both paths of exporting to OSW (one for simulation, one for saving parameter values) into one path that supports both
* Add dummy measure custom_export_params, which holds parameters used by the GUI for selection values, which do not have a function for the simulation but are required for importing
* Workaround for an issue with encoding of OSW files, which are UTF-8, but do not actually support UTF-8
* Remove measure AddingPhotovoltaics and with it the special handling of running a simulation only for PV as well as parameters for it
* Some cleanup of deprecated code, updates of measure.xml files, etc.

### CODE REVIEW

File: Measures/AddDetailedHVAC/measure.xml [hunk 1]
```diff
@@ -63,6 +87,132 @@
       <required>true</required>
       <model_dependent>false</model_dependent>
     </argument>
+    <argument>
+      <name>hvacSchedFeiertag</name>
```
Comment (TobiasMaile) [inline Measures/AddDetailedHVAC/measure.xml:91]: I think we should change this to Holiday
and likewise Werktag to Weekday and so on

File: Measures/AddingIdealAirLoads/measure.xml [hunk 2]
```diff
@@ -63,6 +63,20 @@
       <required>true</required>
       <model_dependent>false</model_dependent>
     </argument>
+    <argument>
+      <name>hvacSchedFeiertag</name>
```
Comment (TobiasMaile) [inline Measures/AddingIdealAirLoads/measure.xml:67]: Similar comment as above. 
I think we do not use this measure anymore, Should we just remove it?

File: Measures/AddingInternalLoads/measure.rb [hunk 3]
```diff
@@ -43,6 +43,21 @@ def arguments(model)
 	args << OpenStudio::Measure::OSArgument.makeStringArgument("PeopleActivityScheduleSonntag", true)
```
Comment (TobiasMaile) [inline Measures/AddingInternalLoads/measure.rb:43]: Sonntag => Sunday

File: Measures/CustomExportParams/measure.xml [hunk 4]
```diff
@@ -0,0 +1,191 @@
+<measure>
```
Comment (TobiasMaile) [inline Measures/CustomExportParams/measure.xml:1]: Do we need this measure? it does not seem to do anything?

File: README.md [hunk 5]
```diff
@@ -5,8 +5,11 @@ Generische Gebäudesimulation auf Basis von EnergyPlus™.
 
 GenSim erzeugt anhand einfacher Eingabeparameter ein vollständiges EnergyPlus™-Gebäudemodell, simuliert dieses und gibt anschließend Ergebnisse in Form von Lastprofilen und Jahreswerten zurück.
```
Comment (TobiasMaile) [inline README.md:7]: I think these changes need to be cleanup up, it looks like merge comments are still included.

**Review by TobiasMaile — CHANGES_REQUESTED** [review]
I think we should replace the German words in the last arguments, like Sonntag, Werktag, etc. to make it consistent.

**Comment by TobiasMaile** [comment]
Where are the automated tests?

File: Measures/AddDetailedHVAC/measure.xml (hunk 1 above)
Comment (eott-siz) [reply Measures/AddDetailedHVAC/measure.xml:91]: Is adressed in a later commit (branch rename-measure-arguments)

File: Measures/AddingIdealAirLoads/measure.xml (hunk 2 above)
Comment (eott-siz) [reply Measures/AddingIdealAirLoads/measure.xml:67]: The name is adressed in a later commit (branch rename-measure-arguments). The measure is not added to the workflow by default. As far as I can tell we can safely remove the measure. (Added to tasks)

File: Measures/AddingInternalLoads/measure.rb (hunk 3 above)
Comment (eott-siz) [reply Measures/AddingInternalLoads/measure.rb:43]: Is adressed in a later commit (branch rename-measure-arguments)

File: Measures/CustomExportParams/measure.xml (hunk 4 above)
Comment (eott-siz) [reply Measures/CustomExportParams/measure.xml:1]: This is the dummy measure for certain parameters, which are not required for performing a simulation, but are necessary for the import/export functionality. That said, some of these additional parameters are placed in other measure simply due to the history of the code. Perhaps in the future we can move all of these to this measure. (Added to tasks)

File: README.md (hunk 5 above)
Comment (eott-siz) [reply README.md:7]: Adressed in a later commit in branch refactor-fix-import-osw

**Comment by eott-siz** [comment]
> Where are the automated tests?

Do you mean the tests for the measures being called as hooks when pushing as defined in `.github/workflows(ruby.yml`? These have not yet been updated for the new repository and activated on GitHub. (Added to tasks)

**Pull request merged**